src/Network/Probe.php

```diff
@@ -58,26 +59,23 @@ class Probe
 	/**
 	 * Remove stuff from an URI that doesn't belong there
 	 *
-	 * @param string $URI
+	 * @param string $uri
 	 * @return string Cleaned URI
 	 */
-	public static function cleanURI(string $URI)
+	public static function cleanURI(string $uri): string
 	{
 		// At first remove leading and trailing junk
-		$URI = trim($URI, "@#?:/ \t\n\r\0\x0B");
-
-		$parts = parse_url($URI);
+		$uri = trim($uri, "@#?:/ \t\n\r\0\x0B");
 
-		if (empty($parts['scheme'])) {
-			return $URI;
+		$uri = new Uri($uri);
+		if (!$uri->getScheme()) {
+			return $uri->__toString();
```

Comment:
Shouldn't we return `$uri` here?

Reply:
At this point `$uri` is an object and the original function was returning a string so we need to cast it back to string.

Reply:
Yeah, I guess at that moment I realized this - which is another reason for using a different name for the two variables :grin: